File: docs/js/Core1/Service-Desk-2/CustomerArrival.js
class CustomerArrival extends eVENT {
  constructor({ occTime, delay, serviceDesk}) {
    super({occTime, delay});
    this.serviceDesk = serviceDesk;
    this.customer = null;  // is assigned on occurrence
  }
  onEvent() {
    var followupEvents=[];
    // create new customer object
    this.customer = new Customer({arrivalTime: this.occTime});
    // push new customer to the queue
    this.serviceDesk.waitingCustomers.push( this.customer);
    // update statistics
    sim.stat.arrivedCustomers++;
    if (this.serviceDesk.waitingCustomers.length > sim.stat.maxQueueLength) {
      sim.stat.maxQueueLength = this.serviceDesk.waitingCustomers.length;
    }
    // if the service desk is not busy
    if (this.serviceDesk.waitingCustomers.length === 1) {
      followupEvents.push( new CustomerDeparture({
        delay: ServiceDesk.serviceDuration(),
        serviceDesk: this.serviceDesk
      }));
    }
    return followupEvents;
  }
  createNextEvent() {
    return new CustomerArrival({
      delay: CustomerArrival.recurrence(),
      serviceDesk: this.serviceDesk
    });
  }
  static recurrence() {
    return math.getUniformRandomNumber( 1, 6);
  }
}

File: Python/Core1/service_desk_2/CustomerArrival.py
import sys, os

module_path = os.path.abspath('framework/')
sys.path.insert(1, module_path)
from oes_foundation import eVENT

module_path = os.path.abspath('lib/')
sys.path.insert(1, module_path)
import util,Customer,CustomerDeparture,ServiceDesk
from ServiceDesk import ServiceDesk
class CustomerArrival(eVENT):
    def __init__(self, sim, serviceDesk, occTime=None, delay=None):
        super().__init__( sim, occTime, delay)
        self.serviceDesk = serviceDesk
        self.customer = None;  # is assigned on occurrence

    def onEvent( self, sim):
        followupEvents = []
        # create new customer object
        self.customer = Customer.Customer(sim,arrivalTime=self.occTime)
        
        # push new customer to the queue
        self.serviceDesk.waitingCustomers.append( self.customer)

        # update statistics
        sim.stat["arrivedCustomers"]=sim.stat["arrivedCustomers"]+1

        if (len(self.serviceDesk.waitingCustomers) > sim.stat["maxQueueLength"]):
            sim.stat["maxQueueLength"] = len(self.serviceDesk.waitingCustomers)
        
        if (len(self.serviceDesk.waitingCustomers) >= 1) :
            followupEvents.append( CustomerDeparture.CustomerDeparture(sim,delay = ServiceDesk.serviceDuration(),serviceDesk= self.serviceDesk))
        
        return followupEvents

    @staticmethod
    def recurrence(): return util.getUniformRandomNumber(1, 6)
    
    def createNextEvent(self, sim):
        delay = CustomerArrival.recurrence()
        serviceDesk=self.serviceDesk
        return CustomerArrival( sim, serviceDesk, delay)
     




       


